python/lsst/ip/diffim/imageDecorrelation.py

```diff
@@ -52,188 +54,211 @@ def computeClippedImageStats(im):
+        @param subtractedExposure[in] the subtracted exposure produced by
+        ip_diffim.ImagePsfMatchTask.subtractExposures()
+        @param psfMatchingKernel an (optionally spatially-varying) PSF matching kernel produced
+        by ip_diffim.ImagePsfMatchTask.subtractExposures()
+        @param log an input pexLog for outputting messages
+
+        @return the decorrelated diffim and decorrelation correction kernel (which may be ignored)
+        The returned decorrelated diffim has an updated PSF as well.
+
+        @note the subtractedExposure is NOT updated
+        @note Here we currently convert a spatially-varying matching kernel into a constant kernel,
+        just by computing it at the center of the image (tickets DM-6243, DM-6244).
+        @note We are also using a constant accross-the-image measure of sigma (sqrt(variance)) to compute
+        the decorrelation kernel.
+        @note Still TBD (ticket DM-6580): understand whether the convolution is correctly modifying
+        the variance plane of the new subtractedExposure.
+        """
+        self.log.info("Starting.")
+        kimg = None
+        try:
+            if psfMatchingKernel.isSpatiallyVarying():
+                spatialKernel = psfMatchingKernel
+                kimg = afwImage.ImageD(spatialKernel.getDimensions())
+                bbox = subtractedExposure.getBBox()
+                xcen = (bbox.getBeginX() + bbox.getEndX()) / 2.
+                ycen = (bbox.getBeginY() + bbox.getEndY()) / 2.
+                spatialKernel.computeImage(kimg, True, xcen, ycen)
+        except:  # Not a spatially-varying kernel (usually not true)
+            kimg = psfMatchingKernel.computeImage()
+
+        # Compute the images' sigmas (sqrt of variance)
+        _, _, sig1squared = computeClippedImageStats(templateExposure.getMaskedImage())
+        sig1 = np.sqrt(sig1squared)
+
+        _, _, sig2squared = computeClippedImageStats(exposure.getMaskedImage())
+        sig2 = np.sqrt(sig2squared)
+
+        corrKernel = self.computeDecorrelationKernel(kimg.getArray(), sig1=sig1, sig2=sig2)
+        fcorrKernel = self.fixEvenKernel(corrKernel)
+        self.log.info("Convolving.")
+        correctedExposure, corrKern = self.doConvolve(subtractedExposure, fcorrKernel,
+                                                      use_scipy=self.config.useScipy)
+        self.log.info("Updating correctedExposure and its PSF.")
+
+        # Compute the subtracted exposure's updated psf
+        psf = subtractedExposure.getPsf().computeImage().getArray()
+        psfc = self.computeCorrectedDiffimPsf(fcorrKernel, psf, sig1=sig1, sig2=sig2)
+        psfcI = afwImage.ImageD(psfc.shape[0], psfc.shape[1])
+        psfcI.getArray()[:, :] = psfc
+        psfcK = afwMath.FixedKernel(psfcI)
+        psfNew = measAlg.KernelPsf(psfcK)
+        correctedExposure.setPsf(psfNew)
+        self.log.info("Complete.")
+
+        return correctedExposure, corrKern
```

Comment:
Please return a pipeBase.Struct. This is our standard for all task run methods, and any other method that returns multiple items, because it supports adding additional items later without breaking existing code.